src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -29,40 +30,61 @@ const RangeStep = styled.button`
   border: none;
   font-size: 0.8rem;
   cursor: pointer;
-  color: blue;
+  color: ${({ theme }) => theme.primary1};
   padding: 0;
 `
 
-const INVESTMENT_STEPS = [0, 25, 50, 75, 100]
+const INVESTMENT_STEPS = ['0', '25', '50', '75', '100']
 
-export default function InvestOption({ approveData, updateInvestAmount, claim }: InvestOptionProps) {
-  const { currencyAmount, price, cost: maxCost, investedAmount } = claim
+function _scaleValue(maxValue: CurrencyAmount<Currency>, value: string) {
+  // parse percent to string, example 25% -> 4 or 50% -> 2
+  const parsedValue = new Fraction(value, '100')
+
+  // divide maxValue with parsed value to get invest amount
+  return maxValue.multiply(parsedValue).asFraction
+}
+
+export default function InvestOption({ approveData, claim, optionIndex }: InvestOptionProps) {
+  const { currencyAmount, price, cost: maxCost } = claim
+  const { updateInvestAmount } = useClaimDispatchers()
+  const { investFlowData } = useClaimState()
+
+  const investedAmount = useMemo(() => investFlowData[optionIndex].investedAmount, [investFlowData, optionIndex])
+
+  const [percentage, setPercentage] = useState<string>(INVESTMENT_STEPS[0])
 
   const { account } = useActiveWeb3React()
 
   const token = currencyAmount?.currency
 
   const balance = useCurrencyBalance(account || undefined, token)
 
-  const handlePercentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    console.log(event.target.value)
-  }
+  const decimals = balance?.currency?.decimals
 
-  const handleStepChange = (value: number) => {
-    console.log(value)
-  }
+  const handleStepChange = useCallback(
+    (value: string) => {
+      if (!maxCost || !balance) {
+        return
+      }
+
+      const scaledCurrencyAmount = _scaleValue(maxCost, value)
+
+      updateInvestAmount({ index: optionIndex, amount: scaledCurrencyAmount.quotient.toString() })
+      setPercentage(value)
+    },
+    [balance, maxCost, optionIndex, updateInvestAmount]
+  )
 
   const onMaxClick = useCallback(() => {
     if (!maxCost || !balance) {
       return
     }
 
     const amount = maxCost.greaterThan(balance) ? balance : maxCost
-    // store the value as a string to prevent unnecessary re-renders
-    const investAmount = formatUnits(amount.quotient.toString(), balance.currency.decimals)
 
-    updateInvestAmount(claim.index, investAmount)
-  }, [balance, claim.index, maxCost, updateInvestAmount])
+    updateInvestAmount({ index: optionIndex, amount: amount.quotient.toString() })
+    setPercentage(INVESTMENT_STEPS[INVESTMENT_STEPS.length - 1])
```

Comment:
what a elaborate way to just set the percentage to 100%

Reply:
hehe that's my bad :P we did this together on the call